kernel/src/hil/time.rs

```diff
@@ -93,8 +113,33 @@ pub trait Client {
 /// The `Timer` trait models a timer that can notify when a particular interval
 /// has elapsed.
 pub trait Timer: Time {
+    fn set_client(&self, client: &'static TimerClient);
+
     /// Sets a one-shot timer to fire in `interval` clock-tics.
     fn oneshot(&self, interval: u32);
     /// Sets repeating timer to fire every `interval` clock-tics.
     fn repeat(&self, interval: u32);
+
+    // Q(alevy): Implementing this might require an additional, unnecessary field if a repeating
+    // timer is distinguished by having a non-zero value in the reload register. What if the return
+    // value is `Option<u32>` and `None` means it's oneshot, `Some(interval)` means it's repeating.
+    // Side benefit, `is_oneshot` and `is_repeating` can have default implementations.
+    fn interval(&self) -> u32;
```

Comment:
+1, I think I like that interface better

Reply:
Agreed.